metrics.py

```diff
@@ -108,14 +112,18 @@ def calc_change_score(current_network: np.ndarray, indices: tuple):
         return -1 if current_network[indices[0], indices[1]] else 1
 
     @staticmethod
-    @njit
-    def calculate_for_sample(networks_sample: np.ndarray):
+    # @njit
+    def calculate_for_sample(networks_sample: np.ndarray | torch.Tensor):
         """
         Sum each matrix over all matrices in sample
         """
-        n = networks_sample.shape[0]
-        reshaped_networks_sample = networks_sample.reshape(n ** 2, networks_sample.shape[2])
-        return np.sum(reshaped_networks_sample, axis=0)
+        if isinstance(networks_sample, torch.Tensor) and networks_sample.is_sparse:
+            return networks_sample.sum(axis=(0, 1))
+        elif isinstance(networks_sample, np.ndarray):
```

Comment:
This is relevant only if we use numba (otherwise we can use the same syntax as for torch, with a `tuple` parameter for `axis`. I think it is better to unify the syntax anyways, so it would be more readable and we spare the `if`s. Doesn't reshape work the same for torch? if not, we should call `.sum()` two times, each time with a single axis, so both numba and torch will support the syntax

Reply:
Now this is split into two different function - one for the numba decorator (which requires the reshape) and one for the sparse matrices (which aren't supported by numba)

Reply:
numba doesn't require the reshape (it can be done by `networks_sample.sum(axis=0).sum(axis=0)`, which I assume is faster, but we can check)